### PR TITLE
WIP: Add test for cleartokens management command.

### LINKED ISSUE #1065: Write a test for cleartokens management command
**Is your feature request related to a problem? Please describe.**

Code coverage of the cleartokens management command has been 0% since it was created in 2015.

**Describe the solution you'd like**

Write a test for the cleartokens command.

**Describe alternatives you've considered**

It's not core functionality so could continue to have 0% coverage but it would be nice to add.

**Additional context**

See also #969

### PR DESCRIPTION
**Pull request by n2ygk**
Fixes #1065

## Description of the Change
Add test for cleartokens command

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [x] unit-test added
- [ ] documentation updated
- [x] `CHANGELOG.md` updated (only for user relevant changes)
- [x] author name in `AUTHORS`

### CODE REVIEW

File: tests/test_commands.py [hunk 1]
```diff
@@ -125,3 +135,60 @@ def test_validation_failed_message(self):
...
+                expires=earlier,
+                redirect_uri="https://localhost/redirect",
+            )
+            GrantModel.objects.create(
+                user=user,
+                code="new grant code {}".format(i),
+                application=app,
+                expires=later,
+                redirect_uri="https://localhost/redirect",
+            )
```
Comment (Andrew-Chen-Wang) [inline tests/test_commands.py:183]: lint not running?

File: tests/test_commands.py [hunk 2]
```diff
@@ -125,3 +135,60 @@ def test_validation_failed_message(self):
...
+            client_type="confidential",
+            authorization_grant_type="client-credentials",
+            name="The App",
+        )
+        for i in range(100):
+            old = AccesstokenModel.objects.create(token="old access token {}".format(i), expires=earlier)
+            new = AccesstokenModel.objects.create(token="current access token {}".format(i), expires=later)
+            # make half of these Access Tokens have related Refresh Tokens, which prevent their deletion.
+            if i % 2:
+                RefreshtokenModel.objects.create(
```
Comment (Andrew-Chen-Wang) [inline tests/test_commands.py:158]: Use bulk_create. Use list comprehensions with RefreshtokenModel(...) and run RefreshtokenModel.objects.bulk_create The order in id increment should remain the same.

File: tests/test_commands.py [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
+    get_access_token_model,
+    get_application_model,
+    get_grant_model,
+    get_refresh_token_model,
+)
 
 
 Application = get_application_model()
+AccesstokenModel = get_access_token_model()
+RefreshtokenModel = get_refresh_token_model()
```
Comment (Andrew-Chen-Wang) [inline tests/test_commands.py:20]:
```suggestion
RefreshTokenModel = get_refresh_token_model()
```
camel case

File: tests/test_commands.py [hunk 4]
```diff
@@ -1,14 +1,24 @@
...
+from oauth2_provider.models import (
+    get_access_token_model,
+    get_application_model,
+    get_grant_model,
+    get_refresh_token_model,
+)
 
 
 Application = get_application_model()
+AccesstokenModel = get_access_token_model()
```
Comment (Andrew-Chen-Wang) [inline tests/test_commands.py:19]:
```suggestion
AccessTokenModel = get_access_token_model()
```
camel case

File: tests/test_commands.py [hunk 5]
```diff
@@ -125,3 +135,60 @@ def test_validation_failed_message(self):
...
+                expires=later,
+                redirect_uri="https://localhost/redirect",
+            )
+
+    def test_command_clear_tokens(self):
+        self.assertEqual(AccesstokenModel.objects.count(), 200)
+        self.assertEqual(RefreshtokenModel.objects.count(), 100)
+        self.assertEqual(GrantModel.objects.count(), 200)
+        call_command(
+            "cleartokens",
```
Comment (Andrew-Chen-Wang) [inline tests/test_commands.py:190]: lint?

**Review by Andrew-Chen-Wang — CHANGES_REQUESTED** [review]

**Comment by n2ygk** [comment]
On second thought clear_tokens is already tested in test_models so maybe just `pragma: no cover`?

Might want to enhance the tests there though....

**Comment by n2ygk** [comment]
@auvipy please stop pushing commits

File: tests/test_commands.py (hunk 1 above)
Comment (n2ygk) [reply tests/test_commands.py:183]: It's what `black` appears to want. I always run `tox -e flake8` before committing. Also the pre-commit hooks do that as well I believe.

File: tests/test_commands.py (hunk 5 above)
Comment (n2ygk) [reply tests/test_commands.py:190]: I blame `black` although this was me:-)

**Comment by n2ygk** [comment]
Closing in lieu of the simpler approach mentioned above.

**Pull request closed without merging**